### ISSUE TITLE
How to upgrade VirtualBox

### ISSUE THREAD

**Issue by Yamakasi** [vboxme/Portable-VirtualBox#53]
Is there some howto how to upgrade the virtualbox version I have installed using the PortableVbox package ?

**Comment by p060477** [comment]
do you know that you won't be able to use bridged network,but only nat,
and no bi-directional drag and drop..??
..theese two BUGS are still there since so many years..
as for them to Runar...Buvick...
what a shame...

**Comment by Yamakasi** [comment]
Can you please stop hijacking everyones topics ? thanks!

**Comment by p060477** [comment]
when runar finally FIXED the BUGS i will be very very glad
cheers!
;)

**Comment by Megachip** [comment]
@Yamakasi this should be happens automatically (if github/vbox.me stay up2date). It checks [vboxinstall.ini line 4](https://github.com/vboxme/Portable-VirtualBox/blob/master/data/settings/vboxinstall.ini#L4) for new updates. If there is a [newer version](https://github.com/vboxme/Portable-VirtualBox/blob/master/data/settings/settings.ini#L43). If there is a newer one, it will download and install it. But looks like this URL is not touched since 21-Mar-2012.

You could setup your own update mirror:
just put the [vboxinstall.ini](https://github.com/vboxme/Portable-VirtualBox/blob/master/data/settings/vboxinstall.ini) into the folder and rename it to vboxinstall.dat and create a 'update.dat' with the following content:
```
[version]
key=5122
[starter]
key=6410
```
Cause I have no time/possibilities to check the update process (atm) I will not update these values. But feel free to share you experience here. Especially if the upgrade from 4.1.10 (latest version on vbox.me) to 5.1.22 works without issues (if yes, possible someone could update the values there).

**Comment by p060477** [comment]
"works without issues "...:

1) not possible to use network bridged SINCE YEARS
2) no bi-directional drag n drop function also SINCE YEARS

is it enough?
:)

**Comment by Megachip** [comment]
@p060477 i didn't mean existing issues. I think your problem is easy to fix, just use Windows 7 or below and everything will be fine.

**Comment by p060477** [comment]
you suggest to use an OLD operative system?
do you know that after it , have come:
-win 8
and then
-win 10
SINCE YEARS

**Comment by Megachip** [comment]
@p060477 correct. Noone said ever that PVB is fully compatible with any os version. So stop hijacking every issue or paid someone to fix your problem.
P.S. a lot of ATMs working with windows NT (4.0 or below), so why not using win 7 ^^ ;)

**Comment by p060477** [comment]
"P.S. a lot of ATMs working with windows NT (4.0 or below), so why not using win 7 ^^ ;) "...
..correct..for the joy of thieves..
is so strange that PVB is STOPPED to win 7
a SO older of YEARS
so many FREE, so NON PAID, sw are FULLY suitable with
newest vers of WIN 10
:)

**Comment by p060477** [comment]
ps : the problem is NOT only mine..but for the MILLIONS of people already using WIN 8 or WIN 10
:)

**Comment by gusbemacbe** [comment]
@Megachip, does it require admin privileges or rights to update?

**Comment by Megachip** [comment]
@gusbemacbe You need the same rights like to launch PVB, so at least admin priv. The update itself didn't need any Privilegs cause it just replaces the local files. But for launching drivers need to be registered and unregistered.

**Comment by gusbemacbe** [comment]
Too bad, I do not use Windows on my Mac, I use VirtualBox to run a virtual machine from my external HD on a college lab computer.

**Comment by runarbu** [comment]
I have made a new binary for the newest version (5.1.26). It is 01:00 in the night her now, so I won't have time to test this much today, but if you want you can download it here: http://files.vbox.me/files/Portable-VirtualBox_v5.1.26-Starter_v6.4.10-Win_all.exe

If everything is fine I will publish it on the web page tomorrow.

**Comment by gusbemacbe** [comment]
Do not publish it now, bitte! Because I tested, it freezed the lab computer, it didn't open, and I couldn't type, I couldn't close the windows. I had to reboot. You need to fix the bugs firstly. 

The most of lab computers use Windows 7 64bits.

**Comment by p060477** [comment]
@runarbu:
does it finally support network bridged?
does it finally support bi-drag n drop?
FOR WIN 10 64 bits obviously
if NOT please WRITE IT CLEARLY on yr vbox.me web page
instead of cheat win 10 billions of customers
CHEERS

**Comment by gusbemacbe** [comment]
Cheat? VBOX is OPEN SOURCE. Will you accuse Atom, Linux and other open source apps of cheating? 

Sei il più stupido dall'Italia per non sapere che cosa è _cheat_ ed _open source_. Sei un troll. I programmatori non sono i tuoi impiegati e schiavi, ragazzo frivolo. Tu non sei capo del Github.

**Comment by p060477** [comment]
@runarbu:
 does it finally support network bridged?
 does it finally support bi-drag n drop?
 FOR WIN 10 64 bits obviously
 if NOT please WRITE IT CLEARLY on yr vbox.me web page
 instead of misleading (is it better..??!! ;)  win 10 billions of customers
 CHEERS

**Comment by Megachip** [comment]
@gusbemacbe Hmm, I've tried the old installer with 5.1.26 and everything worked fine (except the known things). But this was a up2date win10. Have no Win 7 to test :(

**Comment by p060477** [comment]
@Megachip : 
just you are using win 10  : 
PVB
does it finally support network bridged?
 does it finally support bi-dir drag n drop?
 FOR WIN 10 64 bits obviously
 if NOT please WRITE IT CLEARLY on yr vbox.me web page
 so the  win 10 billions of customers won't be misleaded
 CHEERS

**Comment by gusbemacbe** [comment]
@Megachip have you already tried to test on your friends' PCs with Windows 7?

**Comment by Megachip** [comment]
@gusbemacbe Sadly I know noone with Windows 7,8,8.1 :( and have not virtual boxes with this systems anymore (and no time to create them atm). So I think @runarbu should (as you said) wait, until someone confirms that it didn't install on Win7 or someone has the time and the Ressources to test all the different operating versions

**Comment by rdavydov** [comment]
@Megachip Could you specify what are the non-working known things? BTW guys, you have a mess in links and versions on vbox.me: in the Installation instructions it says version _5.1.22_ but links to _5.0.26_ and in the right sidebar it says Newest -> VirtualBox _v5.0.26_ but links, this time correctly, to _5.1.22_.

**Comment by p060477** [comment]
@rdavydov: i reply immediately to yr clear questions:
1)if you have win 8, 8.1 or 10 you won't be able to use:
   network bridged but only NAT
   and you will not have bi directional drag and drop
2) the vbox.me page is a nightmare..they at least write clearly what i've just write here above in point 1)
    it should be helpful not to have billions of people misleaded
3)i'm writing here theese simple things since 2015..but NO useful answers till now..

goodluck!!

**Comment by jlfarinha** [comment]
@Yamakasi - I know this is a workaround, but worked well for me - I managed to upgrade to the latest version of VirtualBox without loosing any settings nor VirtualMachines. 

And I also managed to install the respective "Extension Pack" for that same version I upgraded to.

**Still:** to be on the safe side, before you start, make a copy of the folder where you have _Portable-VirtualBox.exe_, including all its contents and subfolders - if anything goes wrong you can copy everything back.

Here's how it worked for me:

1. Download the latest VirtualBox version from [http://virtualbox.org](http://virtualbox.org) and save it to the folder where you have _Portable-VirtualBox.exe_

2. Download the respective "Virtual Box Extension Pack" from the same page, and also save it to folder where you have _Portable-VirtualBox.exe_

3. Quit the Windows processes _Portable-VirtualBox.exe_ and _VirtualBox.exe_ - if necessary log off and back on, to make sure all running programs are closed

4. Go to the folder where you have _Portable-VirtualBox.exe_ and rename the folder _app64_ (*) to _app64.old_

5. Start _Portable-VirtualBox.exe_ - it will come up with the installation screen asking if you'd like to download VirtualBox or to chose where the download is located; chose the file downloaded in "Step 1)", above 

6. Go through the regular setup. If all goes well, at the end the new version will launch and you'll have all your VMs and settings intact.


Did it work?

**Comment by jlfarinha** [comment]
@p060477 : are you using Google translate and is it failing on you?

**Comment by gusbemacbe** [comment]
@jlfarinha, do not waste time with him. It is an Italian troll. He hijacked every topic and blamed the developers and called them cheaters and thieves.

**Comment by p060477** [comment]
@jlfarinha  : DID yr workaround solved the 2 BIG BUGS still there due to the lack of its developer:
                   1) are you able to use BRIDGED NETW or ONLY NAT ?
                   2) have you a bi-dir Drag n Drop ?

if not you all are only a MESS!!!
or carrying on old so as win 7
cheers!!
;)
:)

**Comment by gusbemacbe** [comment]
@p060477, fuck you and go to hell. You should be banned from GitHub!

**Comment by p060477** [comment]
@LORDgusbemacbe: 
1)are you able to use network bridged and bi dir drag n drop with s.o newer then the oldest win 7?
  if yes i'm a troll...it NO you really are a MESS
2)you should be banned due to yr language..
3) are you try to admit at trinity college in cambridge..??

cheers!
:)
